### ISSUE TITLE
App Lifecycle Improvement

### ISSUE THREAD

**Issue by woodmicha** [anis-campos/network-connectivity-check#2]
Thank you for creating and publishing your solution here.  I've identified one improvement adding the interface `LifecycleObserver` to the App and to invoke `registerConnectivityMonitor()` in the `Lifecylcle.Event.ON_RESUME` callback.  I had to add `implementation 'androidx.lifecycle:lifecycle-process:2.2.0'` to `build.gradle` to support this

The basis for this was the app was losing the ability to monitor the events after it was Paused. 

Thanks again for the crafty kotlin.  I found it super educational on many different aspects of the language while solving a frustrating source for many errors within our app.

**Comment by anis-campos** [comment]
Thanks for the head-up. Would you like to create a PR ? I would gladly review and merge it.

**Comment by woodmicha** [comment]
Sure.  I'll try to get to this over the weekend.  Best, Mike

**Comment by paulocoutinhox** [comment]
it was fixed?

**Comment by anis-campos** [comment]
@prsolucoes surely not, I'm still waiting for the PR

**Comment by sergiofdomingues** [comment]
Does this problem persist?

**Comment by anis-campos** [comment]
> Does this problem persist?

yes, it's still in need of a brave one to fix it. I'm not event working in android projects, so I'm event less incline to fix it now than last year, sadly 😭

**Comment by woodmicha** [comment]
I’ll try to do this next week.  I pointed out the issue and have fixed it
locally.

**Comment by paulocoutinhox** [comment]
Hi @woodmicha,

Can you open a pull-request?

Thanks.
